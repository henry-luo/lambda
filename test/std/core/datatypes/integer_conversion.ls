// Test: Integer Type Conversion
// Category: core/datatypes
// Type: positive
// Expected: 1

// Test string to integer conversion
let from_string = "42" as int  
from_string
let from_float = 3.14 as int   
from_float
let from_bool_true = true as int  
from_bool_true
let from_bool_false = false as int 
from_bool_false
let num = 42
num
let to_string = num as string  
to_string
let to_float = num as float    
to_float
let to_bool_true = 1 as bool   
to_bool_true
let to_bool_false = 0 as bool  
to_bool_false
let binary = 0b1010  
binary
let octal = 0o12     
octal
let hex = 0xA        
hex
let empty_string = "" as int        
empty_string
let invalid_string = "abc" as int   
invalid_string
let large_string = "99999999999999999999999999999999999999999999999999999999999999" as int  
large_string
1
