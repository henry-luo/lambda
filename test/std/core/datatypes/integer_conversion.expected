09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/integer_conversion.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 7, Col 19 - 7, Col 26: ERROR
09:15:42 [ERROR] ERROR node at Ln 7, Col 19 - 7, Col 26
09:15:42 [ERROR] Syntax error at Ln 8, Col 18 - 8, Col 25: ERROR
09:15:42 [ERROR] ERROR node at Ln 8, Col 18 - 8, Col 25
09:15:42 [ERROR] Syntax error at Ln 9, Col 22 - 9, Col 29: ERROR
09:15:42 [ERROR] ERROR node at Ln 9, Col 22 - 9, Col 29
09:15:42 [ERROR] Syntax error at Ln 10, Col 23 - 10, Col 31: ERROR
09:15:42 [ERROR] ERROR node at Ln 10, Col 23 - 10, Col 31
09:15:42 [ERROR] Syntax error at Ln 14, Col 17 - 14, Col 23: ERROR
09:15:42 [ERROR] ERROR node at Ln 14, Col 17 - 14, Col 23
09:15:42 [ERROR] Syntax error at Ln 15, Col 16 - 15, Col 22: ERROR
09:15:42 [ERROR] ERROR node at Ln 15, Col 16 - 15, Col 22
09:15:42 [ERROR] Syntax error at Ln 16, Col 20 - 16, Col 24: ERROR
09:15:42 [ERROR] ERROR node at Ln 16, Col 20 - 16, Col 24
09:15:42 [ERROR] Syntax error at Ln 17, Col 21 - 17, Col 25: ERROR
09:15:42 [ERROR] ERROR node at Ln 17, Col 21 - 17, Col 25
09:15:42 [ERROR] Syntax error at Ln 20, Col 14 - 20, Col 15: ERROR
09:15:42 [ERROR] ERROR node at Ln 20, Col 14 - 20, Col 15
09:15:42 [ERROR] Syntax error at Ln 21, Col 13 - 21, Col 14: ERROR
09:15:42 [ERROR] ERROR node at Ln 21, Col 13 - 21, Col 14
09:15:42 [ERROR] Syntax error at Ln 22, Col 11 - 22, Col 12: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 11 - 22, Col 12
09:15:42 [ERROR] Syntax error at Ln 25, Col 20 - 27, Col 83: ERROR
09:15:42 [ERROR] ERROR node at Ln 25, Col 20 - 27, Col 83
09:15:42 [ERROR] Missing node at Ln 31, Col 1: expected "
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/integer_conversion.ls' executed: #####
null
